server/src-lib/Hasura/Tracing.hs

```diff
@@ -197,13 +200,30 @@ instance MonadTrace m => MonadTrace (ExceptT e m) where
   currentReporter = lift currentReporter
   attachMetadata = lift . attachMetadata
 
+-- | Encode Word64 to 16 character hex string
+word64ToHex :: Word64 -> Text
+word64ToHex randNum = bsToTxt $ Hex.encode numInBytes
+  where numInBytes = BL.toStrict (BIN.encode  randNum)
+
+-- | Decode  16 character hex string to Word64
+-- | TODO: Do we need to check if it has an invalid base16 sequence? 
```

Comment:
Opinion needed on this.

IMHO we should not need this, because we can be sure that the value we recieve from the HTTP headers is actually a proper encoded hex value.

Reply:
In general, we have no control over what's in the headers we receive, so we need to check. We can check that `snd (Hex.decode ...)` is empty, which means that the decoded string had no errors. But this function will need to return a `Maybe Word64`.

Reply:
Fixed it :)